pontoon/machinery/views.py

```diff
@@ -160,34 +164,34 @@ def microsoft_terminology(request):
 
         obj['locale'] = locale
 
-    sources = client.factory.create('ns0:TranslationSources')
-    sources["TranslationSource"] = ['Terms', 'UiStrings']
-
     payload = {
-        'text': text,
-        'from': 'en-US',
-        'to': locale,
-        'sources': sources,
-        'maxTranslations': 5
+        "uuid": uuid4(),
+        "text": text,
+        "to": locale,
+        "max_result": 5
     }
+    template = get_template('machinery/microsoft_terminology.jinja')
+    payload = template.render(payload)
 
     try:
-        r = client.service.GetTranslations(**payload)
+        r = requests.post(url, data=payload, headers=headers)
         translations = []
-
-        if len(r) != 0:
-            for translation in r.Match:
+        xpath = ".//{http://api.terminology.microsoft.com/terminology}"
+        if r.status_code != 200:
+            return HttpResponseBadRequest('Bad Request: Got {code} status code from service'.format(code=r.status_code))
+        root = ET.fromstring(r.content)
+        if len(root.find(xpath + 'GetTranslationsResult')):
+            for translation in root.find(xpath + 'GetTranslationsResult'):
                 translations.append({
-                    'source': translation.OriginalText,
-                    'target': translation.Translations[0][0].TranslatedText,
-                    'quality': translation.ConfidenceLevel,
+                    'source': translation.find(xpath + 'OriginalText').text,
+                    'target': translation.find(xpath + 'TranslatedText').text,
+                    'quality': int(translation.find(xpath + 'ConfidenceLevel').text),
                 })
 
             obj['translations'] = translations
-
         return JsonResponse(obj)
 
-    except WebFault as e:
+    except Exception as e:
```

Comment:
Request time-out error, other request related error, xpath error (will be `text` call on NoneType). I don't expect any error here but if Microsoft's translation service will crash, that will make it to crash too. Just wanted to prevent that.